### PR TITLE
Docker in Ansible

### PR DESCRIPTION
**Pull request by mgax**
This patch sets up docker using ansible and pulls in [this dokuwiki image](https://hub.docker.com/r/istepanov/dokuwiki/). For image building, since docker doesn't like chroot, the build script also installs docker in the kitchen bulidbot VM. Currently hardcoded for x86_64. Fixes liquidinvestigations/liquidinvestigations#36 liquidinvestigations/liquidinvestigations#24.

Remaining work:
* Integrate dokuwiki authentication with `liquid-core` (liquidinvestigations/liquidinvestigations#38)
* Port the dokuwiki image to `arm64` (liquidinvestigations/liquidinvestigations#41)

### CODE REVIEW

File: ansible/roles/common/tasks/main.yml [hunk 1]
```diff
@@ -33,6 +33,7 @@
     - postgresql-contrib
     - postgresql-server-dev-all
     - python-psycopg2 # so that ansible can manage postgres users and dbs
+    - python-pip
```
Comment (gabriel-v) [inline ansible/roles/common/tasks/main.yml]: Is this used anywhere?

File: ansible/roles/dokuwiki/tasks/docker.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+---
+- name: Create folder
+  file:
+    path: /opt/dokuwiki
+    state: directory
+    mode: 0755
+
+- name: Write compose file
+  template:
+    src: ../templates/docker-compose.yml
```
Comment (gabriel-v) [inline ansible/roles/dokuwiki/tasks/docker.yml]: Why is a relative path needed? Maybe we could rearrange things

File: ansible/roles/dokuwiki/templates/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+  dokuwiki:
+    image: istepanov/dokuwiki:2.0
+    volumes:
+      - dokuwiki:/var/dokuwiki-storage
+    ports:
+      - 61906:80
+
+volumes:
+  dokuwiki:
```
Comment (gabriel-v) [inline ansible/roles/dokuwiki/templates/docker-compose.yml]: This makes me feel weird. Is there no other way of writing this without a trailing `:`?

File: ansible/roles/liquid-core/tasks/main.yml [hunk 4]
```diff
@@ -35,7 +35,7 @@
     owner: liquid
 
 - name: Collect django static files
-  shell: ../venv/bin/python ./manage.py collectstatic --noinput
+  shell: /opt/liquid-core/venv/bin/python ./manage.py collectstatic --noinput
```
Comment (gabriel-v) [inline ansible/roles/liquid-core/tasks/main.yml]: It might look cleaner, but was this change needed?

File: ansible/roles/liquid-core/tasks/main.yml [hunk 5]
```diff
@@ -23,7 +23,7 @@
     state: directory
 
 - name: Run migrations
-  shell: ../venv/bin/python ./manage.py migrate
+  shell: /opt/liquid-core/venv/bin/python ./manage.py migrate
```
Comment (gabriel-v) [inline ansible/roles/liquid-core/tasks/main.yml]: Same as below: do you need to change this?

File: ansible/roles/liquid-core/templates/local.py [hunk 6]
```diff
@@ -19,3 +19,4 @@
 
 HOOVER_APP_URL = 'http://hoover.{{ liquid_domain }}'
 HYPOTHESIS_APP_URL = 'http://hypothesis.{{ liquid_domain }}'
+DOKUWIKI_APP_URL = 'http://dokuwiki.{{ liquid_domain }}'
```
Comment (gabriel-v) [inline ansible/roles/liquid-core/templates/local.py]: Did you document this config parameter in liquid-core?
Did you include the DokuWiki link in the liquid-core home page?
I think it'd be cleaner if we'd merge this PR and the liquid-core one at the same time.

File: ansible/roles/dokuwiki/templates/docker-compose.yml (hunk 3 above)
Comment (mgax) [inline ansible/roles/dokuwiki/templates/docker-compose.yml]: Is this better?
```yaml
volumes:
  dokuwiki: {}
```

File: ansible/roles/liquid-core/templates/local.py (hunk 6 above)
Comment (mgax) [inline ansible/roles/liquid-core/templates/local.py]: Eh, I just pushed [the patch](https://github.com/liquidinvestigations/core/commit/bf5df3d), it was straightforward. The other `*_APP_URL`s are not documented either.

File: ansible/roles/dokuwiki/templates/docker-compose.yml (hunk 3 above)
Comment (gabriel-v) [inline ansible/roles/dokuwiki/templates/docker-compose.yml]: Yes, much better. Thanks.

File: ansible/roles/liquid-core/templates/local.py (hunk 6 above)
Comment (gabriel-v) [inline ansible/roles/liquid-core/templates/local.py]: Okay, good.

**Comment by mgax** [comment]
> Also, it would be nice to have docs regarding the build_docker_images var and image_host_docker.yml.a

I agree, but I can't think of a good place to document that variable. It's set in all 3 playbooks and will be used in any role that needs to set up docker images.

File: ansible/roles/liquid-core/tasks/main.yml (hunk 5 above)
Comment (mgax) [inline ansible/roles/liquid-core/tasks/main.yml]: Probably not.
